subsys/bluetooth/controller/ticker/ticker.c

```diff
@@ -791,17 +811,44 @@ void ticker_worker(void *param)
 		 */
 		if (ticker->ticks_slot != 0U &&
 		   (slot_reserved || ticker_resolve_collision(node, ticker))) {
+#if defined(CONFIG_BT_TICKER_EXT)
+			if (ticker->ext_data &&
+			    ticker->ext_data->ticks_slot_window != 0U &&
+			    ticker->ext_data->reschedule_state ==
+			    TICKER_RESCHEDULE_STATE_NONE &&
+			   (ticker->lazy_periodic <= ticker->lazy_current)) {
+				/* Mark node for re-scheduling in ticker_job */
+				ticker->ext_data->reschedule_state =
+					TICKER_RESCHEDULE_STATE_PENDING;
+			} else if (ticker->ext_data) {
+				/* Mark node as not re-scheduling */
+				ticker->ext_data->reschedule_state =
+					TICKER_RESCHEDULE_STATE_NONE;
+			}
+#endif /* CONFIG_BT_TICKER_EXT */
 			ticker->lazy_current++;
+
 			if ((ticker->must_expire == 0U) ||
-			    (ticker->lazy_periodic >= ticker->lazy_current)) {
-				/* Not a must-expire case or this is programmed
-				 * latency. Skip this ticker node
+			    (ticker->lazy_periodic >= ticker->lazy_current) ||
+			    TICKER_RESCHEDULE_PENDING(ticker)) {
```

Comment:
We always skip timeout callback if rescheduling has become pending. Must-expire is handled at rescheduled time- also if there's no free slot found.